app/services/order-service.js

```diff
@@ -186,5 +186,37 @@ export default ApiService.extend({
     this.get("store").pushPayload(data);
     cb(data);
     return data;
+  },
+
+  async createGoodsDetails(orderId, params) {
+    const url = `/goodcity_requests`;
+    const data = await this.POST(url, {
+      goodcity_request: {
+        quantity: params.quantity,
+        order_id: orderId,
+        package_type_id: params.packageTypeId
+      }
+    });
+    this.get("store").pushPayload(data);
+    return data;
+  },
+
+  async updateGoodsDetails(orderId, params) {
```

Comment:
can we create a separate method for the common code.
```
_makeRequest(action, url , params) {
   //move common logic here
   return data;
}
```
In createGoodsDetails
```
async createGoodsDetails(orderId, params) {
    const url = `/goodcity_requests`;
    return _makeRequest('POST', url, params)
 }
```
Similary for Update as well

Just a suggestion.